datalake/src/test/java/com/sequenceiq/datalake/service/sdx/SdxEventsServiceTests.java

```diff
@@ -0,0 +1,213 @@
+                .thenReturn(mockPage);
+        when(eventV4Endpoint.getCloudbreakEventListByStack(any(), any(), any(), any()))
+                .thenReturn(List.of(createCloudbreakEventV4Response()));
+
+        List<CDPStructuredEvent> result = sdxEventsService.getDatalakeAuditEvents(RESOURCE_CRN,
+                Collections.singletonList(StructuredEventType.NOTIFICATION), TEST_PAGE, TEST_SIZE);
+
+        assertNotNull(result);
+        assertEquals(2, result.size());
+        assertEquals(2, result.stream().filter(notification -> notification.getOperation().getEventType().equals(StructuredEventType.NOTIFICATION)).count());
+    }
+
+    @Test
+    void testGetAuditEventsWithCBAndDlEventsSort() {
+        Page<CDPStructuredEvent> mockPage = mock(Page.class);
+
+        when(mockPage.getContent()).thenReturn(List.of(createTestCDPStructuredEvent(StructuredEventType.NOTIFICATION, 5L),
+                createTestCDPStructuredEvent(StructuredEventType.NOTIFICATION, 2L)));
+        when(mockCdpStructuredEventDBService.getPagedEventsOfResources(eq(List.of(StructuredEventType.NOTIFICATION)), any(), any()))
+                .thenReturn(mockPage);
+        when(eventV4Endpoint.getCloudbreakEventListByStack(any(), any(), any(), any()))
+                .thenReturn(List.of(createCloudbreakEventV4Response(1L)));
+
+        List<CDPStructuredEvent> result = sdxEventsService.getDatalakeAuditEvents(RESOURCE_CRN,
+                Collections.singletonList(StructuredEventType.NOTIFICATION), TEST_PAGE, TEST_SIZE);
+
+        assertNotNull(result);
+        assertEquals(3, result.size());
+        assertEquals(3, result.stream().filter(notification -> notification.getOperation().getEventType().equals(StructuredEventType.NOTIFICATION)).count());
+        assertTrue(result.get(0).getOperation().getTimestamp() < result.get(1).getOperation().getTimestamp());
+        assertTrue(result.get(1).getOperation().getTimestamp() < result.get(2).getOperation().getTimestamp());
+    }
+
+    private CloudbreakEventV4Response createCloudbreakEventV4Response() {
+        return createCloudbreakEventV4Response(1L);
+    }
+
+    private CloudbreakEventV4Response createCloudbreakEventV4Response(Long timestamp) {
+        CloudbreakEventV4Response cloudbreakEventV4Response = new CloudbreakEventV4Response();
+        cloudbreakEventV4Response.setEventTimestamp(timestamp);
+        cloudbreakEventV4Response.setClusterName("somename");
+        cloudbreakEventV4Response.setEventMessage("testing");
+        return cloudbreakEventV4Response;
+    }
+
+    private CDPStructuredEvent createTestCDPStructuredEvent(StructuredEventType type) {
+        if (type.equals(StructuredEventType.NOTIFICATION)) {
+            return createCDPStructuredNotificationEvent(0L);
+        } else {
+            return createCDPStructuredFlowEvent(0L);
+        }
+
+    }
+
+    private CDPStructuredEvent createTestCDPStructuredEvent(StructuredEventType type, Long timestamp) {
```

Comment:
`StructuredEventType type` is always `NOTIFICATION`

Reply:
I wanted this to be generic to support other event types as well.